File: Noisie/Noisie/Infrastructure/Dto/RoArtistInfo.m
//
//  RoArtistInfo.m
//  Noisie
//
//  Created by William Swanson on 11/18/11.
//  Copyright (c) 2011 Secret Squirrel Products. All rights reserved.
//

#import "RoArtistInfo.h"

@implementation RoArtistInfo

@synthesize ArtistId;
@synthesize Website;
@synthesize Biography;
@synthesize ArtistImage;
@synthesize BandMembers;
@synthesize TopAlbums;
@synthesize SimilarArtists;

- (void) dealloc {
    self.ArtistId = nil;
    self.Biography = nil;
    self.Website = nil;
    self.ArtistImage = nil;
    self.BandMembers = nil;
    self.TopAlbums = nil;
    self.SimilarArtists = nil;
    
    [super dealloc];
}
@end

File: Noise.Infrastructure/RemoteDto/RoArtistInfo.cs
using System.Runtime.Serialization;

namespace Noise.Infrastructure.RemoteDto {
	[DataContract]
	public class RoArtistInfo {
		[DataMember]
		public long				ArtistId { get; set; }
		[DataMember]
		public string			Website { get; set; }
		[DataMember]
		public string			Biography { get; set; }
		[DataMember]
		public string			ArtistImage { get; set; }
		[DataMember]
		public string[]			BandMembers { get; set; }
		[DataMember]
		public string[]			TopAlbums { get; set; }
		[DataMember]
		public string[]			TopTracks { get; set; }
		[DataMember]
		public long[]			TopTrackIds { get; set; }
		[DataMember]
		public string[]			SimilarArtists { get; set; }

		public RoArtistInfo() {
			Website = string.Empty;
			Biography = string.Empty;
			ArtistImage = string.Empty;
			BandMembers = new string[0];
			TopAlbums = new string[0];
			TopTracks = new string[0];
			TopTrackIds = new long[0];
			SimilarArtists = new string[0];
		}
	}
}
